src/classifier.jl

```diff
@@ -8,6 +8,7 @@ mutable struct NeuralNetworkClassifier{B,F,O,L} <: MLJModelInterface.Probabilist
     lambda::Float64 # regularization strength
     alpha::Float64  # regularizaton mix (0 for all l2, 1 for all l1)
     optimiser_changes_trigger_retraining::Bool
+    gpu::Bool       # Train on gpu
```

Comment:
Let make things consistent as @ablaom as suggested by making use of `ComputationalResources.jl`. Maybe something in the lines of.
```julia
function use_gpu(resource)
 if resource isa Cudalibs
     return true
else if resource isa CPU1
      return false
else
      throw(ArgumentError("Resource not supported"))
end
end#you might want to create a `clean!` method

mutable struct NeuralNetworkClassifier{B,F,O,L} <: MLJModelInterface.Probabilistic
    builder::B
    finaliser::F
    optimiser::O    # mutable struct from Flux/src/optimise/optimisers.jl
    loss::L         # can be called as in `loss(yhat, y)`
    epochs::Int          # number of epochs
    batch_size::Int # size of a batch
    lambda::Float64 # regularization strength
    alpha::Float64  # regularizaton mix (0 for all l2, 1 for all l1)
    optimiser_changes_trigger_retraining::Bool
    gpu::AbstractResource       # Train on gpu
end

NeuralNetworkClassifier(; builder::B   = Short()
              , finaliser::F = Flux.softmax
              , optimiser::O = Flux.Optimise.ADAM()
              , loss::L      = Flux.crossentropy
              , epochs       = 10
              , batch_size   = 1
              , lambda       = 0
              , alpha        = 0
              , optimiser_changes_trigger_retraining = false
              , gpu  = CPU1()
              ) where {B,F,O,L} =
                  NeuralNetworkClassifier{B,F,O,L}(builder
                                       , finaliser
                                       , optimiser
                                       , loss
                                       , epochs
                                       , batch_size
                                       , lambda
                                       , alpha
                                       , optimiser_changes_trigger_retraining
                                       , gpu
                                       )

```
then 
```julia
chain, history = fit!(chain, optimiser, model.loss,
                          model.epochs, model.lambda,
                          model.alpha, verbosity, data, model.gpu)
```
becomes
```julia
chain, history = fit!(chain, optimiser, model.loss,
                          model.epochs, model.lambda,
                          model.alpha, verbosity, data, use_gpu(model.gpu))
```